kratos/factories/register_convergence_criteria_factory.cpp

```diff
@@ -0,0 +1,54 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "factories/base_factory.h"
+#include "spaces/ublas_space.h"
+
+// Convergence criterias
+#include "solving_strategies/convergencecriterias/convergence_criteria.h"
+#include "solving_strategies/convergencecriterias/displacement_criteria.h"
+#include "solving_strategies/convergencecriterias/residual_criteria.h"
+#include "solving_strategies/convergencecriterias/and_criteria.h"
+#include "solving_strategies/convergencecriterias/or_criteria.h"
+
+namespace Kratos
+{
+void RegisterConvergenceCriterias()
+{
+    typedef TUblasSparseSpace<double> SpaceType;
+    typedef TUblasDenseSpace<double> LocalSpaceType;
+
+    typedef DisplacementCriteria<SpaceType,  LocalSpaceType> DisplacementCriteriaType;
+    typedef ResidualCriteria<SpaceType,  LocalSpaceType> ResidualCriteriaType;
+    typedef And_Criteria<SpaceType,  LocalSpaceType> And_CriteriaType;
+    typedef Or_Criteria<SpaceType,  LocalSpaceType> Or_CriteriaType;
+
+    //NOTE: here we must create persisting objects for the linear solvers
+    static DisplacementCriteriaType msDisplacementCriteria;
```

Comment:
See @philbucher the default constructors are used here (and the others RegisterXXX)